src/main/webapp/WEB-INF/component/activity-helper.jsp

```diff
@@ -62,22 +68,23 @@ public String formatCreationTime(Instant time) {
 // Reusable method that returns the layout for the user.
 public String formatUserName(UUID userId) {
     User user = UserStore.getInstance().getUser(userId);
-    if(user == null) return errorMessage;
-    return user.getName();
+    if(user == null) return "UserNotFound";
+    String userLink = String.format("/users/%s", user.getName());
+    return String.format("<a href=\"%s\">%s</a>", userLink, user.getName());
 }
 
 // Reusable method that returns the layout for the conversation.
 public String formatConversation(UUID conversationId) {
     Conversation conversation = ConversationStore.getInstance().getConversation(conversationId);
-    if(conversation == null) return errorMessage;
+    if(conversation == null) return "ConversationNotFound";
     String conversationLink = String.format("/chat/%s", conversation.getTitle());
     return String.format("<a href=\"%s\">%s</a>", conversationLink, conversation.getTitle());
 }
 
 // Reusable method that returns the layout for the message.
 public String formatMessage(UUID messageId) {
     Message message = MessageStore.getInstance().getMessage(messageId);
-    if(message == null) return errorMessage;
+    if(message == null) return "MessageNotFound";
```

Comment:
Same as above, this method should either assume the entity exist or simply throw an exception if it doesn't.